src/lib/dapp/index.ts

```diff
@@ -107,54 +122,39 @@ export class DAppConnector {
         projectId: this.projectId,
         metadata: this.dAppMetadata,
       })
-      const existingSessions = await this.checkPersistedState()
-
-      if (existingSessions.length) {
-        await this.onSessionConnected(existingSessions.pop()!)
+      DAppSigner.initialize(this.walletConnectClient)
 
-        while (existingSessions.length) {
-          this.disconnect(existingSessions.pop()!.topic)
-        }
-      }
+      const existingSessions = this.checkPersistedState()
 
-      this.walletConnectClient.on('session_event', (event) => {
-        // Handle session events, such as "chainChanged", "accountsChanged", etc.
-        alert('There has been a session event!')
-        console.log(event)
-      })
-
-      this.walletConnectClient.on('session_update', ({ topic, params }) => {
-        // Handle session update
-        alert('There has been a update to the session!')
-        const { namespaces } = params
-        const _session = this.walletConnectClient!.session.get(topic)
-        // Overwrite the `namespaces` of the existing session with the incoming one.
-        const updatedSession = { ..._session, namespaces }
-        // Integrate the updated session state into your dapp state.
-        console.log(updatedSession)
-      })
+      if (existingSessions) this.signers = existingSessions.flatMap(this.createSigners)
 
       this.walletConnectClient.on('session_delete', (pairing) => {
-        console.log(pairing)
         this.signers = this.signers.filter((signer) => signer.topic !== pairing.topic)
         this.disconnect(pairing.topic)
-        // Session was deleted -> reset the dapp state, clean up from user session, etc.
-        console.log('Dapp: Session deleted by wallet!')
       })
 
       this.walletConnectClient.core.pairing.events.on('pairing_delete', (pairing) => {
-        // Session was deleted
-        console.log(pairing)
         this.signers = this.signers.filter((signer) => signer.topic !== pairing.topic)
         this.disconnect(pairing.topic)
-        console.log(`Dapp: Pairing deleted by wallet!`)
-        // clean up after the pairing for `topic` was deleted.
       })
     } finally {
       this.isInitializing = false
     }
   }
 
+  public checkPersistedState() {
+    if (!this.walletConnectClient) {
+      throw new Error('WalletConnect is not initialized')
+    }
+    return this.walletConnectClient.session.getAll()
+  }
+
+  public getSigner(accountId: AccountId): DAppSigner {
+    const signer = this.signers.find((signer) => signer.getAccountId().equals(accountId))
+    if (!signer) throw new Error('Signer is not found for this accountId')
```

Comment:
nit: Returning undefined should be ok if not found. The dApp is likely better off handling an undefined result than try/catching this exception.

Reply:
Not agree